bgpd/bgp_vty.c

```diff
@@ -12282,11 +12282,16 @@ void bgp_vpn_policy_config_write_afi(struct vty *vty, struct bgp *bgp,
 {
 	int indent = 2;
 
-	if (bgp->vpn_policy[afi].rmap_name[BGP_VPN_POLICY_DIR_FROMVPN])
-		vty_out(vty, "%*simport vrf route-map %s\n", indent, "",
-			bgp->vpn_policy[afi]
+	if (bgp->vpn_policy[afi].rmap_name[BGP_VPN_POLICY_DIR_FROMVPN]) {
+		if (listcount(bgp->vpn_policy[afi].import_vrf))
+			vty_out(vty, "%*simport vrf route-map %s\n", indent, "",
+				bgp->vpn_policy[afi]
 				.rmap_name[BGP_VPN_POLICY_DIR_FROMVPN]);
-
+		else
+			vty_out(vty, "%*sroute-map vpn import %s\n", indent, "",
+				bgp->vpn_policy[afi]
+				.rmap_name[BGP_VPN_POLICY_DIR_FROMVPN]);
+	}
```

Comment:
If I'm reading the code right, these two commands (`import vrf route-map` and `route-map vpn import`) are changing the same data structure `bgp->vpn_policy[afi].rmap[BGP_VPN_POLICY_DIR_FROMVPN]`.

Thinking from the perspective of the new northbound API, retrofitting these commands to the new model will be problematic. Is there a reason why we can't combine them into one single command to remove the redundancy?

Reply:
good point Renato.
currently, there are some checks that do not permit using vpn command when vrf command is used.
I think we should speak to  @paulzlabn

Reply:
send paul mail via paulz@labn.net

Reply:
To be honest I don't see how this helps in any way. We still have two commands editing the same data structure. Before we would display `import vrf route-map` in the running configuration even when we entered the `route-map vpn import` command. Now it's the other way round: we'll display `route-map vpn import` even if we entered the `import vrf route-map` command, *except* if `import vrf` is also configured. For me it seems like we're changing one problem by another one, _frr-reload.py_ will still have issues given its current limitations.